Markdown/Markdown.sublime-syntax

```diff
@@ -1426,45 +1422,40 @@ contexts:
     - include: image-inline
     - include: image-ref
   link-title:
-    - match: \s*(')
-      captures:
-        1: punctuation.definition.string.begin.markdown
+    - match: \'
+      scope: punctuation.definition.string.begin.markdown
       set:
         - meta_scope: string.other.link.description.title.markdown
         - match: \'
           scope: punctuation.definition.string.end.markdown
           pop: true
-        - match: ^\s*$\n?
-          scope: invalid.illegal.non-terminated.link-title.markdown
-          pop: true
-    - match: \s*(")
-      captures:
-        1: punctuation.definition.string.begin.markdown
+        - include: non-terminated-link-title
+    - match: \"
+      scope: punctuation.definition.string.begin.markdown
       set:
         - meta_scope: string.other.link.description.title.markdown
         - match: \"
           scope: punctuation.definition.string.end.markdown
           pop: true
-        - match: ^\s*$\n?
-          scope: invalid.illegal.non-terminated.link-title.markdown
-          pop: true
-    - match: \s*(\()
-      captures:
-        1: punctuation.definition.string.begin.markdown
+        - include: non-terminated-link-title
+    - match: \(
+      scope: punctuation.definition.string.begin.markdown
       set:
         - meta_scope: string.other.link.description.title.markdown
         - match: \)
           scope: punctuation.definition.string.end.markdown
           pop: true
-        - match: ^\s*$\n?
-          scope: invalid.illegal.non-terminated.link-title.markdown
-          pop: true
-    - match: \s*
+        - include: non-terminated-link-title
+    - match: $|(?=\S)
       pop: true
-  after-link-title:
-    - match: '(.*)$\n?'
-      captures:
-        1: invalid.illegal.expected-eol.markdown
+  non-terminated-link-title:
+    - match: ^\s*$\n?
+      scope: invalid.illegal.non-terminated.link-title.markdown
+      pop: true
+  link-ref-def-expect-end:
+    - meta_scope: meta.link.reference.def.markdown
+    - match: .*$
```

Comment:
This will match trailing whitespace. I know it matches the previous behavior, but I don't think whitespace actually matters here. It certainly doesn't on https://spec.commonmark.org/dingus/.

Would be a good oppurtunity to change this here.

Reply:
Sure.